ffi/src/lib.rs

```diff
@@ -5,6 +5,10 @@
 #[macro_use]
 extern crate tracing;
 
+#[cfg(feature = "dpapi")]
+#[deny(unsafe_op_in_unsafe_fn)]
+#[deny(clippy::undocumented_unsafe_blocks)]
```

Comment:
praise: Excellent initiative!

note: I think it’s fine to use `#[warn]` instead of `#[deny]` because we deny warnings in the CI anyway.

thought: I was going to suggest that we [put these in the Cargo.toml](https://github.com/Devolutions/IronRDP/blob/2a98f8be3e92e3e3e7e91eb2b3149d8399153fef/Cargo.toml#L78-L160), but at this point I guess we would have too much to fix at once in the whole ffi crate, right? 😢 

suggestion: Other lints I suggest when dealing with unsafe code:

- invalid_reference_casting
- clippy::multiple_unsafe_ops_per_block
- clippy::transmute_ptr_to_ptr
- clippy::as_ptr_cast_mut
- clippy::cast_ptr_alignment
- clippy::fn_to_numeric_cast_any
- clippy::ptr_cast_constness

Reply:
> note: I think it’s fine to use #[warn] instead of #[deny] because we deny warnings in the CI anyway.

done

> but at this point I guess we would have too much to fix at once in the whole ffi crate, right? 😢

right, but it's work that needs to be done. We must refactor our FFI if we want to use the 2024 edition.

> The [unsafe_op_in_unsafe_fn](https://doc.rust-lang.org/nightly/rustc/lints/listing/allowed-by-default.html#unsafe-op-in-unsafe-fn) lint now warns by default. ([src](https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html)).

There is not much work in the `winscard` module (because some lints were enabled from the start), but a lot of refactoring in the `sspi` module

Reply:
> Do you prefer adding the lints I mentioned now in a separate PR?

yes. I don't want to mix changes 😅. I'd like to do it after the `dev/dpapi` branch merge